shell/platform/android/io/flutter/view/ResourceExtractor.java

```diff
@@ -242,7 +251,7 @@ private boolean extractUpdate(File dataDir) {
         for (String asset : mResources) {
             String resource = null;
             ZipEntry entry = null;
-            if (asset.endsWith(".so")) {
+            if (asset.endsWith(".so") && Build.VERSION.SDK_INT > Build.VERSION_CODES.KITKAT_WATCH) {
```

Comment:
Maybe it would be better to guard the for loop with this, and in the else block just get it from `"lib/" + asset;` or whatever the appropriate non-ABI aware thing is for lower APIs

Reply:
Here again with mixing SDK checks and app logic.